### PR TITLE
sysdata module: fix amd cpus without Tdie

### PR DESCRIPTION
**Pull request by Compunctus**
1) Never divide by zero in _get_cputemp
2) Add (k10temp-pci-00c3, Tctl) as a sensor - for AMD cpus like 4700U
   which lack Tdie
3) Only set self.lm_sensors if both sensor and chip match - this way we won't set Tdie as a sensor for amd cpus lacking it.

### CODE REVIEW

**Comment by Compunctus** [comment]
should also fix my 5800x.
newer 5xxx cpus also output ccd temps... maybe they should be summed up and divided, just like on intel.

```
   "k10temp-pci-00c3":{
      "Adapter": "PCI adapter",
      "Tctl":{
         "temp1_input": 32.625
      },
      "Tccd1":{
         "temp3_input": 35.250
      }
```

**Comment by lasers** [comment]
I have made a PR https://github.com/ultrabug/py3status/pull/2082 too.

**Comment by Compunctus** [comment]
tl;dr will test your PR tomorrow. This one can be closed if it works.

Yeah, i wanted to make as few changes as possible, so i discarded the array idea early on.
Anyway, both approaches will work - will test your PR tomorrow.

I think that you should port the "don't divide by zero" part to #2082. The absence of this check leads to meaningless exceptions.

Will play around with averaging per-ccd temps later.

**Comment by lasers** [comment]
We could give it `None` or `0` with try/except but `sensor_total` would be `0` either way.

We got rid of many try/except and one-off statements in many modules because it was not needed for most part and this sometimes hides the real issues from the users for a while.

Also, because py3status started exposing exception messages on the bar...along with mouse clicks to display/hide/make it go away... 

In order to take advantage of this new exceptions-on-the-bar feature, we had to remove meaningless try/excepts, one-off statements, hardcoded errors that we can't hide with mouse clicks, and errors that would plainly hide (i.e. print "") without telling the users, etc... 

Those had to go so the real exceptions can be seen on the bar... and we get real exception issues from the users instead of slightly vague "This prints zero now" issues... plus... we can throw our own errors nicely too instead of faking red text errors.

**Comment by Compunctus** [comment]
agreed, let's keep it as is.
Tested you PR, it works, closing that one.
